### DISCUSSION TITLE
Customize Database Driver

### DISCUSSION THREAD

**Discussion by gogl92** [tailflow/laravel-orion#283] [Ideas]
As a developer I'd like to be able to modify certain classes so I can use different data sources like NoSQL databases and other database engines.

Driver and/or `standard` classes should be configurable, like

```
return [
    'namespaces' => [
        'models' => 'App\\Models\\',
        'controllers' => 'App\\Http\\Controllers\\',
        'drivers' => 'App\\Drivers\\' \\ Orion\\Drivers by default
    ],
    'driver' => 'mongodb', //standard by default
    'auth' => [
        'guard' => 'api',
    ],
   ...
];
```

This will help to override methods that have SQL specific keywords like `applySearchingToQuery`

May fix
https://github.com/tailflow/laravel-orion/issues/228

**Comment by alexzarbn** [comment]
Hi @gogl92,

Thank you for raising the issue. Sounds quite useful - I will leave the issue open to see if there is further interest in this.
